src/Core/Services/Implementations/OrganizationSponsorshipService.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Threading.Tasks;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Bit.Core.Services
+{
+    public class OrganizationSponsorshipService : IOrganizationSponsorshipService
+    {
+        private const string FamiliesForEnterpriseTokenName = "FamiliesForEnterpriseToken";
+        private const string TokenClearTextPrefix = "BWOrganizationSponsorship_";
+
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationRepository _organizationRepository;
+        private readonly IUserRepository _userRepository;
+        private readonly IPaymentService _paymentService;
+        private readonly IMailService _mailService;
+
+        private readonly IDataProtector _dataProtector;
+
+        public OrganizationSponsorshipService(IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+            IOrganizationRepository organizationRepository,
+            IUserRepository userRepository,
+            IPaymentService paymentService,
+            IMailService mailService,
+            IDataProtectionProvider dataProtectionProvider)
+        {
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _organizationRepository = organizationRepository;
+            _userRepository = userRepository;
+            _paymentService = paymentService;
+            _mailService = mailService;
+            _dataProtector = dataProtectionProvider.CreateProtector("OrganizationSponsorshipServiceDataProtector");
+        }
+
+        public async Task<bool> ValidateRedemptionTokenAsync(string encryptedToken)
+        {
+            if (!encryptedToken.StartsWith(TokenClearTextPrefix))
+            {
+                return false;
+            }
+
+            var decryptedToken = _dataProtector.Unprotect(encryptedToken[TokenClearTextPrefix.Length..]);
```

Comment:
Does this provide any benefit over making it more obvious to use that they work this way? Ideally the `TokenClearTextPrefix` would be the first token which identifies what it is.

Reply:
Which is how `TokenIsValid` works. I think it would be nice with some parser that validates the structure and spits out a DO

Reply:
@MGibson1 Would you like me to switch to a token that the first part is the type of token? We still can't use `TokenIsValid` without a refactor though. But this isn't a thing we can really change down the road as easily without having multiple versions of this token.

Reply:
We decided to keep it as is. I've added a tech debt item to Asana to clean up the token logic project wide.